crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs

```diff
@@ -166,490 +162,536 @@ impl OptChaining {
+                        Expr::Member(MemberExpr {
                             span,
-                            op: op!("delete"),
-                            arg: expr.alt,
-                        })),
-                        ..expr
-                    }
-                    .into();
-                }
+                            obj: ExprOrSuper::Expr(obj),
+                            prop,
+                            computed,
+                        }) if obj.is_opt_chain() => {
+                            let mut obj = *obj.take();
 
-                Expr::Member(MemberExpr {
-                    span,
-                    obj: ExprOrSuper::Expr(obj),
-                    prop,
-                    computed,
-                }) if obj.is_opt_chain() => {
-                    let obj = obj.opt_chain().unwrap();
-                    let expr = self.unwrap(obj);
+                            self.unwrap(&mut obj);
 
-                    return CondExpr {
-                        span: DUMMY_SP,
-                        alt: Box::new(Expr::Unary(UnaryExpr {
-                            span,
-                            op: op!("delete"),
-                            arg: Box::new(Expr::Member(MemberExpr {
-                                span,
-                                obj: ExprOrSuper::Expr(expr.alt),
-                                prop,
-                                computed,
-                            })),
-                        })),
-                        ..expr
+                            if let Expr::Cond(expr) = obj {
+                                *e = CondExpr {
+                                    span: DUMMY_SP,
+                                    alt: Box::new(Expr::Unary(UnaryExpr {
+                                        span: *span,
+                                        op: op!("delete"),
+                                        arg: Box::new(Expr::Member(MemberExpr {
+                                            span: *span,
+                                            obj: ExprOrSuper::Expr(expr.alt),
+                                            prop: prop.clone(),
+                                            computed: *computed,
+                                        })),
+                                    })),
+                                    ..expr
+                                }
+                                .into();
+                            }
+                        }
+                        _ => {}
                     }
-                    .into();
-                }
-                _ => {}
-            }
-        }
-
-        Expr::Unary(e)
-    }
-
-    /// Only called from [Fold].
-    fn handle_call(&mut self, e: CallExpr) -> Result<CondExpr, Expr> {
-        match e.callee {
-            ExprOrSuper::Expr(callee) if callee.is_opt_chain() => {
-                let callee = callee.opt_chain().unwrap();
-                let expr = self.unwrap(callee);
```

Comment:
Can't we just return `CondExpr` from `unwrap`, by taking all components of it?
I think it wil make code much better.

Reply:
That was I tried initially and somehow couldn't make it work at the moment, maybe now all the refs are replaced it may possible. I'll give another shot and either update PR or leave comment here if it doesn't work.

Reply:
@kdy1 , ok, looks like it was able to make handle_ to return as suggested. Thanks for the tip, PR is updated as suggested.